### DISCUSSION TITLE
Quarkus Application Build Property

### DISCUSSION THREAD

**Discussion by cpmoore** [quarkusio/quarkus#35823] [Q&A]
Hi all,

I know some quarkus configuration properties are locked to a specific value during build to improve startup time. 
My question is if there is any hook to do the same for an application built with quarkus. 

For instance, if you have an annotation and you use reflections to scan all the files in your app with this annoatation, can you cache the value of that list as a property so it is automatically set at startup instead of having to perform the scan during initialization. 

Thanks,
Cody

**Comment by maxandersen** [comment]
Make a quarkus extension?

**Comment by cpmoore** [comment]
This code doesn’t make sense to be an extension. It’s business logic that will not be reused in other projects.
